terraform/main.tf

```diff
@@ -27,18 +54,23 @@ module "database" {
   railway_environment_id = railway_project.this.default_environment.id
 }
 
-module "holesky_ensindexer" {
-  source     = "./modules/ensindexer"
+module "ensindexer" {
+  source   = "./modules/ensindexer"
+  for_each = local.ensindexer_instances
+
   depends_on = [null_resource.health_check]
-  #Indexer specific envs
-  base_domain_name       = local.base_domain_name
-  subdomain_prefix       = "holesky.${local.railway_environment}"
-  ensnode_version        = var.ensnode_version
-  heal_reverse_addresses = local.heal_reverse_addresses
-  ensrainbow_url         = "http://$${{${railway_service.ensrainbow.name}.RAILWAY_PRIVATE_DOMAIN}}:8080"
-  database_schema        = "holeskySchema-${var.ensnode_version}"
-  active_plugins         = local.active_plugins
-  ens_deployment_chain   = local.ens_deployment_chain
+  # Instance-specific configuration
+  subdomain_prefix       = each.value.subdomain_prefix
+  database_schema        = each.value.database_schema
+  active_plugins         = each.active_plugins
+  ens_deployment_chain   = each.ens_deployment_chain
+  heal_reverse_addresses = each.heal_reverse_addresses
```

Comment:
Nice, we have instance specific values composed together with the common settings 🫡